packages/components/organisms/f-form/src/components/_tests/Form.test.js

```diff
@@ -76,17 +96,205 @@ describe('Form', () => {
             });
         });
 
-        describe('onFormSubmit :: ', () => {
-            it('should emit `form-submitting`', () => {
+        describe('isFormValid ::', () => {
+            let touchSpy;
+
+            beforeEach(() => {
+                touchSpy = jest.spyOn($v, '$touch');
+            });
+
+            afterEach(() => {
+                jest.clearAllMocks();
+            });
+
+            it('should call `.touch()` ', () => {
+                // Act
+                const wrapper = shallowMount(FForm, { propsData, mocks: { $v } });
+
+                wrapper.vm.isFormValid();
+
+                // Assert
+                expect(touchSpy).toBeCalled();
+            });
+
+            it('should return `true` if `$v` is valid', () => {
                 // Arrange
-                const wrapper = shallowMount(FForm, { propsData });
+                $v.$invalid = false;
+
+                // Act
+                const wrapper = shallowMount(FForm, { propsData, mocks: { $v } });
+
+                // Assert
+                expect(wrapper.vm.isFormValid()).toBeTruthy();
+            });
+
+            it('should return `false` if `$v` is invalid', () => {
+                // Arrange
+                $v.$invalid = true;
+
+                // Act
+                const wrapper = shallowMount(FForm, { propsData, mocks: { $v } });
+
+                // Assert
+                expect(wrapper.vm.isFormValid()).toBeFalsy();
+            });
+        });
+
+        describe('onFormSubmit ::', () => {
+            let isFormValidSpy;
+            let scrollToFirstInlineErrorSpy;
+            let wrapper;
+
+            beforeEach(() => {
+                isFormValidSpy = jest.spyOn(FForm.methods, 'isFormValid');
```

Comment:
Instead of spying on the method, I think it would be a little cleaner to mock the underlying value of `$v.$invalid`.

Spies are perfect when you want to test that something is being called, but also maintain the original functionality.
As you're changing the functionality as part of the test (by mocking the return value), it's probably not an ideal usage.